TTCN3_Project/CANFilter/com/testingtech/playits/canfilter/CANFilterService.java

```diff
@@ -0,0 +1,216 @@
+  private static final String DEFAULT_SERVER_HOST = "localhost";
+
+  private static Hashtable<String, Car2XEntry> car2xEntries = new Hashtable<>();
+  private static Car2XEntryUpdater entryUpdater;
+  private static TimeoutResponder timeoutResponder;
+
+  /**
+   * Provides tcp server functionality to handle json requests. Upon receiving 
+   * a start request, an entry updater thread which updates openXC / obd2 hash 
+   * table entries is started. Parallel timeout responder tasks periodically
+   * send back updated JSON values from the hash table over a client socket.
+   * 
+   * @param args
+   *            host, port
+   */
+  public static void main(String[] args) {
+    String host = DEFAULT_SERVER_HOST;
+    int port = DEFAULT_SERVER_PORT;
+
+    if (args.length != 2) {
+      System.err
+          .println("[CANFilterService] Usage: java CANFilterService "
+              + "<Server host>  <Server Port Number>");
+      System.err.println("  ... using default server host "
+          + DEFAULT_SERVER_HOST + " default server port "
+          + DEFAULT_SERVER_PORT);
+    } else {
+      host = args[0];
+      port = extractPortNumber(args, 1, DEFAULT_SERVER_PORT);
+    }
+
+    InetSocketAddress address = new InetSocketAddress(host, port);
+    Socket socket = null;
+    ServerSocket serverSocket = null;
+    try {
+      serverSocket = createServerSocket(address);
+      System.out.println("[CANFilterService] Listening on "
+              + serverSocket.getInetAddress().getHostName() + ":"
+              + serverSocket.getLocalPort());
+      socket = serverSocket.accept();
+      timeoutResponder = new TimeoutResponder(socket, car2xEntries);
+      JSONTokener jsonTokener = new JSONTokener(socket.getInputStream());
+      while (jsonTokener.more()) {
+        try {
+          Object nextValue = jsonTokener.nextValue();
+          if (nextValue instanceof JSONObject) {
+            System.out
+                .println("[CANFilterService] Incoming request: "
+                    + nextValue);
+            // TODO maybe add a kill request to exit loop
+            handleRequest((JSONObject) nextValue);
+          } else {
+            System.err
+                .println("[CANFilterService] Could not process json request."
+                    + nextValue);
+          }
+        } catch (JSONException e) {
+          System.err.println("[CANFilterService] Could not process"
+              + " request. " + e.getMessage());
+        }
+      }
+
+    } catch (IOException e) {
+      System.err.println("[CANFilterService] Error listening on port "
+          + address.getPort() + ": " + e);
+    } catch (JSONException e) {
+      System.err
+          .println("[CANFilterService] Error while reading json objects from input stream");
+    } finally {
+      System.out
+          .println("[CANFilterService] No more requests. Closing sockets");
+      close(socket);
+      close(serverSocket);
```

Comment:
Please use "this", when you call methods, which are implemented in the same class.